app/(dashboard)/dashboard/_components/ParticipantsTable/ParticipantsTable.tsx

```diff
@@ -33,17 +39,40 @@ export const ParticipantsTable = ({
     },
   });
 
-  const { mutateAsync: deleteParticipants } =
+  const { mutateAsync: deleteParticipants, isLoading: isDeleting } =
     trpc.participant.delete.byId.useMutation();
 
+  const { mutateAsync: deleteAllParticipants, isLoading: isDeletingAll } =
+    trpc.participant.delete.all.useMutation();
+
   const editParticipant = (identifier: string) => {
     setSeletedParticipant(identifier);
     setShowModal(true);
   };
 
-  const handleDelete = async (data: Participant[]) => {
-    await deleteParticipants(data.map((d) => d.identifier));
+  const handleDelete = (data: ParticipantWithInterviews[]) => {
+    setDeleteAll(false);
+    setParticipantsToDelete(data);
+    setShowAlertDialog(true);
+  };
+
+  const handleDeleteAll = () => {
+    setDeleteAll(true);
+    setParticipantsToDelete(participants);
+    setShowAlertDialog(true);
+  };
+
+  const handleConfirm = async () => {
+    if (deleteAll) {
+      await deleteAllParticipants();
+      await refetch();
+      setShowAlertDialog(false);
+    }
+
+    // Delete selected participants
+    await deleteParticipants(participantsToDelete.map((d) => d.identifier));
     await refetch();
+    setShowAlertDialog(false);
```

Comment:
I noticed here `setShowAlertDialog(false)` is being invoked in either the condition true or false, so maybe we don't need one in line 69.

Reply:
resolved with [ba9fa0e](https://github.com/complexdatacollective/Fresco/pull/25/commits/ba9fa0eff2eaf68da29e6d9373ab3bd9e1233bee)